csv-concat.js

```diff
@@ -78,43 +101,46 @@ module.exports = function(options) {
 
         var scannableFiles = getScannableFiles(files);
 
-        scannableFiles.forEach(function(file) {
+        createDestDir(options.dest);
 
-            if (options.verbose) {
-                console.log("   ");
-                console.log("Reading file: " + file);
-            }
+        try {
+            scannableFiles.forEach(function (file) {
+                logV("   ");
+                logV("Reading file: " + file);
 
-            try {
-                var csvFileParser = new CsvFileParser(file, options, parseCsv, fs);
-                var records = csvFileParser.getRecords();
+                var lines = [];
 
-                if (options.verbose) {
-                    console.log("  - Writing " + records.length + " lines into " + options.dest);
-                }
+                try {
+                    var csvFileParser = new CsvFileParser(file, options, parseCsv, fs);
+                    var records = csvFileParser.getRecords();
+
+                    logV("  - Writing " + records.length + " lines into " + options.dest);
 
-                records.forEach(function (recordIter) {
-                    outputStream.write(recordIter.join(options.delimiter) + '\n');
-                });
-            } catch (e) {
-                if (options.verbose) {
-                    console.log("- Failed parsing a record: " + e);
+                    records.forEach(function (recordIter) {
```

Comment:
Well, given that <code>records</code> is an array already it seems that this foreach is redundant now :D